### ISSUE TITLE
RFE: "include" directive in sssd.conf

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1169]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1165
- Created at 2012-02-05 17:42:04 by [mpizzoli](https://pagure.io/user/mpizzoli)
- Closed as Invalid
- Assigned to nobody
---

As an enhancement request I would like a "include" directive in sssd.conf permitting me to split my conf in N configuration fragments all eventually included in the original sssd.conf.

Thanks
   Marco

### Comments

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-02-05 17:47:26](https://pagure.io/SSSD/sssd/issue/1165#239599)**

Could you describe in greater detail how you would use this feature? Are you looking for the ability to store configuration for different domains in separate configuration files? Or are you looking for the ability to set defaults for your domains in the master sssd.conf and add overrides or addons in another file? Or something else entirely?

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-02-05 17:48:56](https://pagure.io/SSSD/sssd/issue/1165#239600)**

Fields changed

component: SSSD => INI Parser
keywords: include configuration sssd.conf => 
milestone: SSSD 1.9.0 => NEEDS_TRIAGE

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-02-05 17:58:59](https://pagure.io/SSSD/sssd/issue/1165#239601)**

Hi,
Both scenarios you described are interesting.
I was thinking more about the former: the possibility to split configuration based on a logic grouping of directives. A file per domain is a good example of what I was intending/asking.

Your second scenario, the possibility to overwrite directives already specified, is also a lot interesting and would give me a lot of flexibility I was not yet thinking at the moment.
I was not asking this, but I think it would be a good feature enhancement.

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-02-09 16:09:04](https://pagure.io/SSSD/sssd/issue/1165#239602)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.11 beta

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-02-13 18:06:26](https://pagure.io/SSSD/sssd/issue/1165#239603)**

Ticket has been cloned to Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=790113

rhbz:  => [https://bugzilla.redhat.com/show_bug.cgi?id=790113 790113]

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-03-19 09:20:15](https://pagure.io/SSSD/sssd/issue/1165#239604)**

This was requested again by a different user - https://fedorahosted.org/pipermail/sssd-devel/2012-March/009043.html

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-03-19 12:55:40](https://pagure.io/SSSD/sssd/issue/1165#239605)**

I think we need to divide this into two tickets. There really are two different requests going on here.

The original request for this ticket was "the possibility to split configuration based on a logic grouping of directives. A file per domain is a good example of what I was intending/asking."

However, the request made in the email thread was more like my second question. He wants the ability to set up the configuration centrally, but be able to merge "override" values into it from a secondary local file.

I think we probably need to account for both cases, but they deserve separate tickets. I've opened ticket #1264 to track this.

blocking:  => 1264

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-08-16 13:37:29](https://pagure.io/SSSD/sssd/issue/1165#239606)**

Fields changed

proposed_priority:  => Nice to have

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-09-04 23:20:32](https://pagure.io/SSSD/sssd/issue/1165#239607)**

Moving all the features planned for 1.10 release into 1.10 beta.

milestone: SSSD 1.11 beta => SSSD 1.10 beta

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2012-12-20 23:33:42](https://pagure.io/SSSD/sssd/issue/1165#239608)**

Fields changed

selected:  => Not need

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2013-01-02 15:32:08](https://pagure.io/SSSD/sssd/issue/1165#239609)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2013-10-25 14:14:44](https://pagure.io/SSSD/sssd/issue/1165#239610)**

Fields changed

cc: marco.pizzoli@gmail.com => marco.pizzoli@gmail.com, Michael.Gliwinski@henderson-group.com
changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
review:  => 0

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2013-10-29 23:47:29](https://pagure.io/SSSD/sssd/issue/1165#239611)**

There is actually no plan to support "include" semantics in ding-libs used by sssd to read INI files. Rather SSSD can potentially merge different configs coming from different files. Ding-libs supports that. It is only the question of defining the rules and using the merge capabilities of ding-libs.

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2014-02-06 17:05:48](https://pagure.io/SSSD/sssd/issue/1165#239612)**

Putting up to NEEDS_TRIAGE so we can discuss the priority on the next meeting.

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2014-02-06 17:06:09](https://pagure.io/SSSD/sssd/issue/1165#239613)**

Fields changed

milestone: SSSD 1.13 beta => NEEDS_TRIAGE

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2014-02-06 17:10:00](https://pagure.io/SSSD/sssd/issue/1165#239614)**

Fields changed

keywords:  => localaccounts

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2014-02-14 02:32:02](https://pagure.io/SSSD/sssd/issue/1165#239615)**

The #include option is not something we want to do. The reasons for that are:
    - With "include" option you need to specify what you include i.e. you need to modify sssd.conf first. We think that the feature should not require explicit include statement because it somewhat defeats the purpose of being able to augment the SSSD configuration without changing main config file.
    - Include option would be a spacial option that parser would have to recognize in the format that is different from everything else. It is sort of a completely new construct grammar wise.

We created a new ticket #2247 that supersedes this one. 

resolution:  => wontfix
status: new => closed

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2014-07-03 02:30:17](https://pagure.io/SSSD/sssd/issue/1165#239616)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.13 beta

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2016-04-18 18:58:51](https://pagure.io/SSSD/sssd/issue/1165#239617)**

Fields changed

design:  => https://fedorahosted.org/sssd/wiki/DesignDocs/ding-libs/INIConfigMerge
mark:  => 0
sensitive:  => 0

---

**Comment from [mpizzoli](https://pagure.io/user/mpizzoli) at [2017-02-24 15:00:33](https://pagure.io/SSSD/sssd/issue/1165#239618)**

**Metadata Update from @mpizzoli**:
- Issue marked as blocked by: #1264
- Issue set to the milestone: SSSD 1.13.1
